### PR TITLE
Apply  MaxConcurrentCalls depdnding on number of processors. Fixes #27

### LINKED ISSUE #27: Apply defaults based on environment capabilities and SKU
Currently, default configuration values are applied without taking the hosting environment into consideration. In order to correctly apply default configuration settings that take advantage of hosting capabilities (additional cores, memory, etc.), we need to take those factors into consideration.

This should apply to all SKUs, except consumption, where we need to ensure we maintain the same behavior we have today.

### PR DESCRIPTION
**Pull request by alrod**

### CODE REVIEW

File: src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Config/ServiceBusOptions.cs [hunk 1]
```diff
@@ -29,6 +29,13 @@ public ServiceBusOptions()
             };
 
             SessionHandlerOptions = new SessionHandlerOptions(ExceptionReceivedHandler);
+
+            // Set configuration for non-dynamic SKUs
+            string skuValue = Environment.GetEnvironmentVariable(Constants.AzureWebsiteSku);
+            if (!string.Equals(skuValue, Constants.DynamicSku, StringComparison.OrdinalIgnoreCase))
+            {
+                MessageHandlerOptions.MaxConcurrentCalls *= Environment.ProcessorCount;
+            }
```
Comment (brettsam) [inline src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Config/ServiceBusOptions.cs]: Are we doing this in response to an issue? Is this a recommendation from SB?

Comment (mathewc) [reply src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Config/ServiceBusOptions.cs]: Don't we have a tracking item for this that you can link to this bug? I think the work item across all triggers is to use more intelligent defaults across all our triggers, to improve performance.

What about SessionHandlerOptions? What is the default for PrefetchCount - should we also default that?

File: test/Microsoft.Azure.WebJobs.Extensions.ServiceBus.Tests/Config/ServiceBusOptionsTests.cs [hunk 2]
```diff
@@ -28,7 +28,7 @@ public ServiceBusOptionsTests()
         public void Constructor_SetsExpectedDefaults()
         {
             ServiceBusOptions config = new ServiceBusOptions();
-            Assert.Equal(16, config.MessageHandlerOptions.MaxConcurrentCalls);
+            Assert.True(config.MessageHandlerOptions.MaxConcurrentCalls >= 16);
```
Comment (mathewc) [inline test/Microsoft.Azure.WebJobs.Extensions.ServiceBus.Tests/Config/ServiceBusOptionsTests.cs]: Recommend making this check stronger and compute the expected value in the test based on processor count. For the E2E test as well.

File: src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Config/ServiceBusOptions.cs (hunk 1 above)
Comment (alrod) [reply src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Config/ServiceBusOptions.cs]: tracking item is #27. `SessionHandlerOptions` doesn't have `PrefetchCount` but it have `MaxConcurrentSessions`. We don't want to change default `MaxConcurrentSessions` depending on core count.

Comment (mathewc) [reply src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Config/ServiceBusOptions.cs]: Ok

File: test/Microsoft.Azure.WebJobs.Extensions.ServiceBus.Tests/Config/ServiceBusOptionsTests.cs (hunk 2 above)
Comment (mathewc) [reply test/Microsoft.Azure.WebJobs.Extensions.ServiceBus.Tests/Config/ServiceBusOptionsTests.cs]: Just fix these two test issues, and the code looks good to me

**Pull request merged**